atomicapp/providers/kubernetes.py

```diff
@@ -38,227 +44,243 @@ class KubernetesProvider(Provider):
+                the default localhost:8080/api end-point.
+
+            Default configuration:
+                If no paramaters have been provided, as well as no --provider-config,
+                use the default .kube/config location and load the associated configuration.
+            '''
 
-        if self.dryrun:
-            # Testing env does not have kubectl in it
-            return "/usr/bin/kubectl"
+            config_location = os.path.join(Utils.getRoot(), Utils.getUserHome().strip('/'), '.kube/config')
 
-        test_paths = ['/usr/bin/kubectl', '/usr/local/bin/kubectl']
-        if self.config.get("provider_cli"):
-            logger.info("caller gave provider_cli: " + self.config.get("provider_cli"))
-            test_paths.insert(0, self.config.get("provider_cli"))
+            if self.config_file:
+                self.api = Client(KubeConfig.from_file(self.config_file), "kubernetes")
+            elif self._check_required_params():
+                self.api = Client(self._from_required_params(), "kubernetes")
+            elif not os.path.isfile(config_location):
+                self.config["provider-api"] = self.provider_api
+                self.api = Client(self._from_required_params(), "kubernetes")
+            else:
+                self.api = Client(KubeConfig.from_file(config_location), "kubernetes")
+
+            # Check if the namespace that the app is being deployed to is available
+            self._check_namespaces()
+
+    def _check_required_params(self):
+        '''
+        This checks to see if any parameters associated to the Kubernetes provider are passed
+        '''
+
+        # Initialize the values
+        result = {PROVIDER_API_KEY: self.provider_api,
+                  PROVIDER_AUTH_KEY: self.provider_auth,
+                  PROVIDER_TLS_VERIFY_KEY: self.provider_tls_verify,
+                  PROVIDER_CA_KEY: self.provider_ca}
+
+        # Get values from the loaded answers.conf / passed CLI params
+        for k in result.keys():
+            result[k] = self.config.get(k)
+
+        if bool([a for a in result.values() if a is None]):
+            logger.debug("No required params have been passed")
+            return False
+
+        return True
+
+    def _from_required_params(self):
+        '''
+        Create a default configuration from passed environment parameters. Only PROVIDER_API_KEY
+        is *required*. Token may be blank.
+        '''
+
+        # Initialize the values
+        result = {PROVIDER_API_KEY: self.provider_api,
+                  PROVIDER_AUTH_KEY: self.provider_auth,
+                  PROVIDER_TLS_VERIFY_KEY: self.provider_tls_verify,
+                  PROVIDER_CA_KEY: self.provider_ca}
+
+        # Get values from the loaded answers.conf / passed CLI params
+        for k in result.keys():
+            result[k] = self.config.get(k)
+
+        # Check that the required parameter is passed. If not, error out.
+        for k in [PROVIDER_API_KEY]:
```

Comment:
i see now where we are checking that only provider api key is set.. and it is the only one required.. we need to have some coordination between `_from_required_params` and `_check_required_params` since they both "check required params"